### PR TITLE
Fiddling around with a hire form / data model

### PR DESCRIPTION
**Pull request by AndyMoreland**
Lots of trash in here generated while I was poking around at hasura migrations & playing with apollo. Just posting this in case you feel like looking at it.

Data model for the table is also mostly just rand() -- wanted to have something notion to play with but it doesn't handle tenants / is missing the 'updated_at' trigger, etc.

### CODE REVIEW

File: hasura/migrations/default/1621895901253_add_employees_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+create table employees (
+			 id uuid primary key,
```
Comment (AndyMoreland) [inline hasura/migrations/default/1621895901253_add_employees_table/up.sql:2]: @osdiab these fields need to get changed substantially (and include the FKs for which tenant the employee belongs to etc) but just wanted to have something to play with.

File: src/components/default-layout/header.tsx [hunk 2]
```diff
@@ -13,8 +15,22 @@ const headerCss = [
...
+  const { user, error, isLoading } = useUser();
+
+  return (
+    <header css={headerCss}>
+      <h1>Spinach</h1>
+      <div>
+        {isLoading ? (
+          <span>Loading...</span>
+        ) : user ? (
+          <Link href="/api/auth/logout">Logout</Link>
```
Comment (AndyMoreland) [inline src/components/default-layout/header.tsx:28]: The start of a top menu or something ...

File: src/pages/employer/dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  const columns = useMemo<Column<UnArray<typeof employees>>[]>(
+    () => [
+      {
+        Header: "Id",
+        accessor: "id",
+      },
+      {
+        Header: "First Name",
+        accessor: "first_name", // accessor is the "key" in the data
```
Comment (AndyMoreland) [inline src/pages/employer/dashboard.tsx:31]: @osdiab so btw one can actually inspect the "GetEmployeesDocument" object and pull all of the fields that are returned in the query -- I intend to noodle around with generating some of this a little more automatically.

File: hasura/migrations/default/1621895901253_add_employees_table/up.sql (hunk 1 above)
Comment (osdiab) [reply hasura/migrations/default/1621895901253_add_employees_table/up.sql:2]: cool, yeah we can deal with tenants after in another migration

File: package.json [hunk 4]
```diff
@@ -74,6 +74,7 @@
     "eslint-plugin-react-hooks": "^4.2.0",
     "eslint-plugin-unicorn": "^32.0.1",
     "graphqurl": "^0.3.3",
+    "hasura-cli": "alpha",
```
Comment (osdiab) [inline package.json:77]: bleeding edge!

**Review by osdiab — COMMENTED** [review]
Wondering whether we need both the base generated graphql file and the ones in the generated folder - I’m guessing the base on is from the gql codegen command, how do they interact?

otherwise lookin like a great start!

**Comment by AndyMoreland** [comment]
@osdiab yeah right before I fell asleep last night I was thinking "hm what parts of the generated code should we actually check in?"

I don't even know if it makes sense to check any of it in since it's all 100% derived from the postgres schema 🤔. Usually I check in the actual graphql schema because it's something that's hand-written rather than generated, but that's not the case here.

**Comment by osdiab** [comment]
I’m ok with checking it in though for practical reasons - if it’s not
there the code doesn’t compile so CI would need to fetch it every time, so
we’d need to configure api keys and might take runtime - I went ahead
earlier and also checked in the caches for yarn 2 (zero-installs) since
that tends to speed things up so I don’t really have an issue with checking
in generated stuff.

Just there seems to be more than one generated graphql schema (I think the
project root one was also generated?) so not sure what is used for what
haha

Omar

**Comment by osdiab** [comment]
fyi made some fancy form inputs and making a few more soon

**Pull request merged**